File: 1/z3.cpp
#include <iostream>
#include <string>
#include <vector>
#include <utility>
#include <algorithm>
#include <ctime>
using namespace std;

pair<string,string> createArray(string a, string b) {
    string up;
    string down;

    vector<int> v;  // 1 -> b ; 0 -> a
    {
        unsigned ones = 0;
        unsigned zero = 0;
        for (int i = 0;; i++) {
            if (ones == b.size() && zero == a.size()) {
                break;
            }
            if (ones == b.size()) {
                v.push_back(0);
                ++zero;
                continue;
            }
            if (zero == a.size()) {
                v.push_back(1);
                ++ones;
                continue;
            }
            int what = rand() % 2;
            if (what) {
                v.push_back(1);
                ++ones;
            }
            else {
                v.push_back(0);
                ++zero;
            }
        }
    }
    //cout << v.size() << endl;
    for (auto x : v) {
        cout << x;
    }
    cout << endl;

    int bit = 0;
    int ait = 0;
    int last = 0;
    for (int i = 0; i < v.size(); i++) {
        if (v[i] != last) {
            up += a[ait++];
            down += b[bit++];
        } else {
            if (!v[i]) {
                up += a[ait++];
                down += '_';
            } else {
                up += '_';
                down += b[bit++];
            }
        }
        last = v[i];
    }
    return make_pair(up, down);
}

int main() {
    string a, b;
    cin >> a >> b;

    srand(time(0));

    for (int i = 0; i < 100; i++) {
        auto x = createArray(a, b);
        cout << x.first << endl;
        cout << x.second << endl;
        cout << endl;
    }
    return 0;
}

File: 1/z3.py
import random
 
def create_array(a, b):
  vec = []
  for i in a:
    vec.append(0)
  for i in b:
    vec.append(1)
  random.shuffle(vec)
   #vec = [1, 0, 0, 1, 0, 0]
  print(vec)
  
  x, y = [], []
  last, ait, bit, i = 1, 0, 0, 0
  for z in vec:
    if z == last:
      if z == 0:
        x.append(a[ait])
        y.append("_")
        ait = ait + 1
      else:
        x.append("_")
        y.append(b[bit])
        bit = bit + 1
    else:
      if z:
        y.append(b[bit])
        bit = bit + 1
      else:
        x.append(a[ait])
        ait = ait + 1
    last = z
    
  if len(x) < len(y):
    x.append("_")
  elif len(y) < len(x):
    y.append("_")
  return "".join(x), "".join(y)
  
def solve():
  a = "HPEW"
  b = "PW"

  for i in range(1, 2):
    x, y = create_array(a, b)
    print(x)
    print(y)

solve()